docs_src/design/adr/security/0019-spiffe.md

```diff
@@ -0,0 +1,338 @@
+but it can be combined with the `spiffe-server` just as well.
+Supported databases are SQLite (default), PostgreSQL, and MySQL.
+Due to startup ordering issues, SQLite will be used.
+
+Workloads are added to the authorization database by running
+the `spiffe-server entry create` CLI to associate an SPID
+to a _selector_.  For example:
+
+```sh
+spire-server entry create -parentID "${local_agent_svid}" -dns edgex-core-data -spiffeID "${svid_service_base}/edgex-core-data" -selector "docker:label:com.docker.compose.service:edgex-core-data"
+```
+
+A snap-based installation of EdgeX would use a `unix:path` or `unix:sha256` selector instead.
+
+There are two extension mechanims for authorization additional workloads:
+
+1. Inject additional scripts into the `spiffe-config` container to authorize additional workloads at startup.
+2. Run the `spire-server entry create` command on-demand to authorize services ad-hoc.
+
+The authorization database is persistent across reboots.
+
+
+## Consequences
+
+This proposal will require addition of several new, optional, EdgeX microservices:
+
+* `security-spiffe-token-provider`, running on the main node
+* `spiffe-agent`, running on the main node and each remote node
+* `spiffe-server`, running on the main node
+* `spiffe-config`, a one-shot service running on the main node
+
+Minor changes will be needed to `security-secretstore-setup` to preserve
+the token-creating-token used by `security-file-token-provider` so that
+it can be used by `security-spiffe-token-provider`.
+
+The startup flow of the framework will be adjusted as follows:
+
+* Bootstrap service (original)
+  * `spiffe-server`
+    * `spiffe-config` (can be combined with `spifee-server`)
+      * `spiffe-agent`
+  * Vault service (original)
+    * Secret store setup service (original)
+      * `security-spiffe-token-provider`
+    * Consul (original)
+    * Postgres (original)
+
+There is no direct dependency between `spiffe-server`
+and any other microservice.
+`security-spiffe-token-provider` requires an SVID
+from `spiffe-agent` and a Vault admin token.
+
+None of these new services will be proxied via the API gateway.
+
+In the future,
+this mechanism may become the default secret store distribution mechanism,
+as it eliminates several secrets volumes used to share secrets between
+`security-secretstore-setup` and various EdgeX microservices.
+
+The EdgeX automation will only configure the SPIFEE agent on the main node.
+Additional nodes can be manually added by the operator
+by obtaining a join token from the main node
+and using it to bootstrap a remote node.
+
+SPIFFE/SPIRE has native support for Kubernetes
+and can distribute the trust bundle via a Kubernetes
+ConfigMap to more easily enable distributed scenarios,
+removing a major roadblock to usage of EdgeX
+in a Kubernetes environment.
+
+### Footprint
+
+#### SPIRE Server
+```
+    69 MB executable, dynamically linked
+   151 MB inside of a Debian-slim container
```

Comment:
Which means there are dependencies required from Debian (and/or Alpine or Ubuntu)?